### ISSUE TITLE
Error when downloading with `thor docs:download`

### ISSUE THREAD

**Issue by mortalis13** [freeCodeCamp/devdocs#1278]
Hi.
I execute the command `thor docs:download html`, and the response is:

    (1/1) HTML FAILED (Errno::ENOENT: No such file or directory - gunzip)

The installation was performed according to the steps from the README, without errors.

    git clone https://github.com/freeCodeCamp/devdocs.git && cd devdocs
    gem install bundler
    bundle install

Specs:  
**Windows 10**  
**Ruby 2.6.6 x86**

---

More info:

Executing `bundle exec rackup` gives:

    Unable to load the EventMachine C extension; To use the pure-ruby reactor, require 'em/pure_ruby'
    Traceback (most recent call last):
            30: from d:/Ruby26/bin/rackup:23:in `<main>'
            29: from d:/Ruby26/bin/rackup:23:in `load'
            28: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/bin/rackup:4:in `<top (required)>'
            27: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:148:in `start'
            26: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:283:in `start'
            25: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:354:in `wrapped_app'
            24: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:219:in `app'
            23: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:319:in `build_app_and_options_from_config'
            22: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:40:in `parse_file'
            21: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:49:in `new_from_string'
            20: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:49:in `eval'
            19: from d:/Docs/GitHub/_other/devdocs/config.ru:in `<main>'
            18: from d:/Docs/GitHub/_other/devdocs/config.ru:in `new'
            17: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:55:in `initialize'
            16: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:55:in `instance_eval'
            15: from d:/Docs/GitHub/_other/devdocs/config.ru:5:in `block in <main>'
            14: from d:/Docs/GitHub/_other/devdocs/config.ru:5:in `require'
            13: from d:/Docs/GitHub/_other/devdocs/lib/app.rb:4:in `<top (required)>'
            12: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler.rb:174:in `require'
            11: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `require'
            10: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `each'
             9: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:69:in `block in require'
             8: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:69:in `each'
             7: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:74:in `block (2 levels) in require'
             6: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:74:in `require'
             5: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thin-1.7.2/lib/thin.rb:7:in `<top (required)>'
             4: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thin-1.7.2/lib/thin.rb:7:in `require'
             3: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/eventmachine-1.2.7-x86-mingw32/lib/eventmachine.rb:8:in `<top (required)>'
             2: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/eventmachine-1.2.7-x86-mingw32/lib/eventmachine.rb:8:in `require'
             1: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/eventmachine-1.2.7-x86-mingw32/lib/rubyeventmachine.rb:2:in `<top (required)>'
    D:/Ruby26/lib/ruby/gems/2.6.0/gems/eventmachine-1.2.7-x86-mingw32/lib/rubyeventmachine.rb:2:in `require': cannot load such file -- 2.6/rubyeventmachine (LoadError)

---

Executing `thor docs:generate javascript --force` gives:

    Fetching redirections...

    [                                                                                                                             ] [0/1] [  0.00%] [00:00] [00:00] [0.00/s][#############################################################################################################################] [1/1] [100.00%] [00:01] [00:00] [0.92/s]Continuing...

    Building internal urls...

    [                                                                                                                             ] [0/1] [  0.00%] [00:00] [00:00] [0.00/s][#############################################################################################################################] [1/1] [100.00%] [00:01] [00:00] [0.93/s]Continuing...

    Traceback (most recent call last):
            51: from d:/Ruby26/bin/thor:23:in `<main>'
            50: from d:/Ruby26/bin/thor:23:in `load'
            49: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/bin/thor:6:in `<top (required)>'
            48: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
            47: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
            46: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
            45: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:128:in `run'
            44: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:29:in `run'
            43: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/runner.rb:42:in `method_missing'
            42: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
            41: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
            40: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
            39: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
            38: from d:/Docs/GitHub/_other/devdocs/lib/tasks/docs.thor:106:in `generate'
            37: from d:/Docs/GitHub/_other/devdocs/lib/tasks/docs.thor:300:in `generate_doc'
            36: from d:/Docs/GitHub/_other/devdocs/lib/docs.rb:100:in `generate'
            35: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/scrapers/url_scraper.rb:142:in `store_pages'
            34: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/scrapers/url_scraper.rb:152:in `with_redirections'
            33: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/scrapers/url_scraper.rb:144:in `block in store_pages'
            32: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/scraper.rb:223:in `store_pages'
            31: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/scraper.rb:233:in `with_internal_urls'
            30: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/scraper.rb:225:in `block in store_pages'
            29: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/doc.rb:107:in `store_pages'
            28: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:85:in `replace'
            27: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:30:in `open'
            26: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:144:in `open_yield_close'
            25: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:85:in `block in replace'
            24: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:87:in `replace'
            23: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:170:in `lock'
            22: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:87:in `block in replace'
            21: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:182:in `track_touched'
            20: from d:/Docs/GitHub/_other/devdocs/lib/docs/storage/abstract_store.rb:87:in `block (2 levels) in replace'
            19: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/doc.rb:108:in `block in store_pages'
            18: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/scraper.rb:74:in `build_pages'
            17: from d:/Docs/GitHub/_other/devdocs/lib/docs/core/instrumentable.rb:15:in `instrument'
            16: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/notifications.rb:168:in `instrument'
            15: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/notifications/instrumenter.rb:29:in `instrument'
            14: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/notifications/instrumenter.rb:44:in `finish_with_state'
            13: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/notifications/fanout.rb:48:in `finish'
            12: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/notifications/fanout.rb:48:in `each'
            11: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/notifications/fanout.rb:48:in `block in finish'
            10: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/notifications/fanout.rb:104:in `finish'
             9: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/activesupport-5.2.3/lib/active_support/subscriber.rb:101:in `finish'
             8: from d:/Docs/GitHub/_other/devdocs/lib/docs/subscribers/progress_bar_subscriber.rb:11:in `running'
             7: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:39:in `write'
             6: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:76:in `to_s'
             5: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:76:in `inject'
             4: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:76:in `each'
             3: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:77:in `block in to_s'
             2: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:92:in `render'
             1: from D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:101:in `render_bar'
    D:/Ruby26/lib/ruby/gems/2.6.0/gems/progress_bar-1.3.0/lib/progress_bar.rb:47:in `ratio': no implicit conversion from nil to integer (TypeError)

---

Don't know if those exceptions are related to the `docs:download` error.  
Maybe the Ruby version is incorrect? The `no implicit conversion from nil to integer` message says that some parameter of the `ratio` method if `nil`, that's weird...  
Thanks.

**Comment by mortalis13** [comment]
Additional info:

When I access the [dl.devdocs.io](https://dl.devdocs.io), it shows me a security error.
In Firefox:

**Potential Security Issue** (**SSL_ERROR_BAD_CERT_DOMAIN**)

![image](https://user-images.githubusercontent.com/10464927/94111201-039be100-fe44-11ea-8e28-6ded1f66c3b7.png)

In Chrome:

**Your connection is not private** (**NET::ERR_CERT_COMMON_NAME_INVALID**)

![image](https://user-images.githubusercontent.com/10464927/94111084-e0713180-fe43-11ea-93b1-088a3349ff20.png)

**Comment by mortalis13** [comment]
I resolved the `No such file or directory - gunzip` error by creating `gunzip.bat` in the `d:\msys64\usr\bin\` directory.
The default script is **Bash**, so it can't be run from Windows.

The contents of the `gunzip.bat`:

    @echo off
    gzip -d %*

The other error, `cannot load such file -- 2.6/rubyeventmachine`, which appears when executing the

    bundle exec rackup

was resolved using the fix from the [issue-1152](https://github.com/freeCodeCamp/devdocs/issues/1152#issuecomment-578187136):

    gem uninstall eventmachine
    gem install eventmachine --platform ruby

---

Although, still cannot execute the `thor docs:generate javascript --force`, gives me the same error.
The browser security errors accessing [dl.devdocs.io](dl.devdocs.io) are not relevant in this case, will open a separate issue.
